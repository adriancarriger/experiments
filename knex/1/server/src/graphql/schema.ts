import { gql } from 'apollo-server';

import customer from './Customer/schema';
import rental from './Rental/schema';

export const base = gql`
  type Query
  type Mutation
`;

export default [base, customer, rental];
