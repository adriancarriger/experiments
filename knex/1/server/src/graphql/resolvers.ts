import { merge } from 'lodash';

import customerResolvers from './Customer/resolvers';
import rentalResolvers from './Rental/resolvers';

export default merge(customerResolvers, rentalResolvers);
