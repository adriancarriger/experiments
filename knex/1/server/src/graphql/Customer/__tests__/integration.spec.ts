import { gql } from 'apollo-server';

import customerResolvers from '../resolvers';
import customerSchema from '../schema';

import { createClient, disconnectDb, Client } from '../../__tests__/graphql.helpers';

const GET_CUSTOMERS = gql`
  query {
    getCustomers {
      first_name
      last_name
      id
    }
  }
`;

describe('Queries', () => {
  let client: Client;

  beforeAll(() => {
    client = createClient([customerSchema], customerResolvers);
  });

  afterAll(disconnectDb);

  it('fetches customers', async () => {
    const { data } = await client.query({ query: GET_CUSTOMERS });
    expect(data).toMatchSnapshot();
  });
});
