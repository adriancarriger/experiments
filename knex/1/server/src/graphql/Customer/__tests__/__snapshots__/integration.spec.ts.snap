// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries fetches customers 1`] = `
Object {
  "getCustomers": Array [
    Object {
      "first_name": "Chris",
      "id": "94",
      "last_name": "Traeger",
    },
    Object {
      "first_name": "Andy",
      "id": "95",
      "last_name": "Dwyer",
    },
    Object {
      "first_name": "April",
      "id": "96",
      "last_name": "Ludgate",
    },
  ],
}
`;
